src/main/scala/junctions/crossing.scala

```diff
@@ -91,3 +90,46 @@ object ClockToSignal {
     c2s.io.signal_out
   }
 }
+
+object LevelCrossing {
+  class SynchronizerBackend[T <: Data](typ: T, sync: Int, _clock: Clock) extends Module(Some(_clock)) {
```

Comment:
For this kind of synchronizer, you should restrict the Data type, to Bool or Vec[Bool]. It does not make sense,  and is misleading, to synchronize a e.g. Bundle or UInt in this way. Force the user to convert it to Bools if that's what they think they want.